### DISCUSSION TITLE
DS18B20 sensor

### DISCUSSION THREAD

**Discussion by flexiti** [HASwitchPlate/openHASP#664] [Show and tell]
**my_custom.cpp**

```
#include "DS18B20.h"
#define ONE_WIRE_BUS       1   (port 1, not used PK2 in my ESP32-4848S040)
OneWire oneWire(ONE_WIRE_BUS);
DS18B20 sensor(&oneWire);
float Tem18B20;

void custom_setup()
{
    sensor.begin();
}

void custom_every_5seconds()
{
if (sensor.isConversionComplete())
            {
         //   Serial.print("Temp:-------> ");
            Tem18B20 = sensor.getTempC();
         //   Serial.println(Tem18B20);

            page=1;   //my page
            id = 40;    // my label
            lv_obj_t* MyTT = hasp_find_obj_from_page_id(page, id);
                  if(MyTT) //return; // object doesn't exist
                        lv_label_set_text_fmt(MyTT,"#F09080 %0.1f°C", Tem18B20 ); 
            }
sensor.requestTemperatures();    // reverse order to sensor.getTempC() , no need to wait
}

```
https://github.com/RobTillaart/DS18B20_RT/tree/master

platformio.ini - add robtillaart/DS18B20@^0.2.2

```
; -- Shared library dependencies in all environments
; Warning : don't put comments after github links => causes infinite download loop
lib_deps =
```
**robtillaart/DS18B20@^0.2.2**
`bblanchon/ArduinoJson@^6.21.5`

:smirk:
`

**Comment by maanr23** [comment]
Hi, do you have the wiring diagram for this example. I am trying to setup the exact temperature sensor with esp32-4848S040

**Comment by flexiti** [comment]
As above
#define ONE_WIRE_BUS       1   (port 1, not used PK2 in my ESP32-4848S040)
+ gnd and vcc

**Reply by maanr23** [reply]
Thanks for the quick reply, is there a any reference guide to find port 1? I wouldn't know where to look on the board

**Comment by flexiti** [comment]
![zd3](https://github.com/user-attachments/assets/b2244ac8-ef61-4069-9bba-6bab3c232828)
![ESP32-WROOM-32-Pinout](https://github.com/user-attachments/assets/65583f60-af5c-47f3-8fe3-c8c34ba1094e)

**Comment by FreeBear-nc** [comment]
Notes below apply to the Guition ESP32-S3 4848S040
If you use GIO1 (relay 3), GPIO2 (relay 2), or GPIO40 (relay 1), you need to remove the zero ohm link labeled R26 (GPIO1), R27 (GPIO2), or R25 (GPIO40). If you don't, the transistor used to turn the relay on may cause problems.

GPIO41 and GPIO42 are available, but these are routed to the TF(SD) slot - The card slot is unused at present within openHASP, so you could use either one for the DS18B20 one wire interface.

GPIO43 and GPIO44 are assigned to the USB interface, so should not be used. (GPIO10 is also unavailable as this is connected to the LCD panel).

**Comment by schnuller123** [comment]
#flexiti

hello, i am new in openhasp, so i have a question
did you run a aditional part of software to use the TempSensor?
your my_custom.cpp i did not understand where you run the part?
sorry for my stupid question????

how can i do this to, normaly i use arduino IDE bur the display of the 4848s04 ist to complex to use it with arduino
so i found openhasp
all is good, but now i will measure a temp so i need an TempSensor like ds18b20
but i dont know how i can do this!
also i have a problem to control the intern GPIO40 to switch a relay with nodered over mqtt
perhaps you can help me a little, i try hasp/plate/command/gpio40 with the value 1/0 or on/off
but it didnt work

**Comment by Serge-IZOARD** [comment]
Hi,
  An idea to integrate a DS18B20 on a bottom of case with a minimal wiring.
  Power +, - and data to the GPIO 41
  In plus I've integrate a radar HLK-LD2410 (GPIO 42) to identify if a person is in front of the unit to manage the backlight and inform on the presence ...
![DS18B20](https://github.com/user-attachments/assets/42b5b078-3e9a-4bb8-a958-43acca3c439b)
![Front](https://github.com/user-attachments/assets/97895f8b-163f-4315-9fac-a8f830afce31)
![Radar](https://github.com/user-attachments/assets/4781b49f-c157-4831-b799-3d13eecde0de)
![Sizes](https://github.com/user-attachments/assets/0c56d764-833e-4471-9399-7ff687283c4f)
![Wiring](https://github.com/user-attachments/assets/eda2abc7-3479-4781-a172-e25a7180d4ef)

**Reply by RealDeco** [reply]
worth to notice there is 4 versions of the ld2410, the one without letter after has no bluetooth, ld2410b has bluetooth, and ld2410c is larger breadboard friendly pin sized and also have bluetooth, and last of is the ld2410s which is not the same sensor at all (not as good), and bluetooth is a good thing with these sensors when finetuning settings ;)
